### PR TITLE
ARROW-4446: [Python] Run Gandiva tests on Windows and Appveyor

### PR DESCRIPTION
**Pull request by shyambits2004**
Disabling the timezone tests till it is fixed for windows.

### CODE REVIEW

File: cpp/src/gandiva/cast_time.cc [hunk 1]
```diff
@@ -42,6 +42,11 @@ void ExportedTimeFunctions::AddMappings(Engine* engine) const {
   engine->AddGlobalMappingForFunc("gdv_fn_time_with_zone",
                                   types->i32_type() /*return_type*/, args,
                                   reinterpret_cast<void*>(gdv_fn_time_with_zone));
+
+  args = {};
+  engine->AddGlobalMappingForFunc("gdv_fn_is_tzdb_configured",
```
Comment (pravindra) [inline cpp/src/gandiva/cast_time.cc]: this isn't being used from IR code. so, why add global mapping ?

File: cpp/src/gandiva/precompiled/time_test.cc [hunk 2]
```diff
@@ -68,29 +68,33 @@ TEST(TestTime, TestCastTimestamp) {
...
-
-  EXPECT_EQ(castTIMESTAMP_utf8(context_ptr, "2000-09-23 9:45:30.920 Unknown/Zone", 35),
-            0);
-  EXPECT_EQ(context.get_error(),
-            "Invalid timestamp or unknown zone for timestamp value 2000-09-23 "
-            "9:45:30.920 Unknown/Zone");
-  context.Reset();
+  if (is_tzdb_configured()) {
+    EXPECT_EQ(
+        castTIMESTAMP_utf8(context_ptr, "2000-09-23 9:45:30.920 Canada/Pacific", 37),
```
Comment (pravindra) [inline cpp/src/gandiva/precompiled/time_test.cc]: can we skip exporting the fn too if tzdb not configured ?

**Review by pravindra — COMMENTED** [review]
please use a different jira for this.

**Comment by wesm** [comment]
I'm just disabling these tests in my patch until this is fixed. is that alright?

**Comment by pravindra** [comment]
> I'm just disabling these tests in my patch until this is fixed. is that alright?

I saw your patch disabling only the windows tests. lgtm

**Comment by xhochy** [comment]
@shyambits2004 Can you rebase on master or would you like help with that?

**Comment by wesm** [comment]
What do you want to do with this patch now?

**Comment by shyambits2004** [comment]
Closing this PR and will track this issue as part of ARROW-4541

**Pull request closed without merging**